src/EnergyPlus/FaultsManager.cc

```diff
@@ -356,33 +408,30 @@ namespace FaultsManager {
             }
 
             // Evaporative cooler check
-            {
-                auto const SELECT_CASE_VAR(state.dataFaultsMgr->FaultsEvapCoolerFouling(jFault_EvapCoolerFouling).EvapCoolerType);
 
-                int EvapCoolerNum;
+            int EvapCoolerNum;
 
-                if (UtilityRoutines::SameString(SELECT_CASE_VAR, "EvaporativeCooler:Indirect:WetCoil")) {
-                    // Read in evaporative cooler is not done yet
-                    if (state.dataEvapCoolers->GetInputEvapComponentsFlag) {
-                        EvaporativeCoolers::GetEvapInput(state);
-                        state.dataEvapCoolers->GetInputEvapComponentsFlag = false;
-                    }
+            if (UtilityRoutines::SameString(state.dataFaultsMgr->FaultsEvapCoolerFouling(jFault_EvapCoolerFouling).EvapCoolerType,
+                                            "EvaporativeCooler:Indirect:WetCoil")) {
+                // Read in evaporative cooler is not done yet
+                if (state.dataEvapCoolers->GetInputEvapComponentsFlag) {
+                    EvaporativeCoolers::GetEvapInput(state);
+                    state.dataEvapCoolers->GetInputEvapComponentsFlag = false;
+                }
 
-                    // Check whether the evaporative cooler  name and type match each other;
-                    EvapCoolerNum =
-                        UtilityRoutines::FindItemInList(state.dataFaultsMgr->FaultsEvapCoolerFouling(jFault_EvapCoolerFouling).EvapCoolerName,
-                                                        state.dataEvapCoolers->EvapCond,
-                                                        &EvaporativeCoolers::EvapConditions::Name);
-                    if (EvapCoolerNum <= 0) {
-                        ShowSevereError(state,
-                                        cFaultCurrentObject + " = \"" + cAlphaArgs(1) + "\" invalid " + cAlphaFieldNames(5) + " = \"" +
-                                            cAlphaArgs(5) + "\" not found.");
-                        state.dataFaultsMgr->ErrorsFound = true;
-                    } else {
-                        // Link the boiler with the fault model
-                        state.dataEvapCoolers->EvapCond(EvapCoolerNum).FaultyEvapCoolerFoulingFlag = true;
-                        state.dataEvapCoolers->EvapCond(EvapCoolerNum).FaultyEvapCoolerFoulingIndex = jFault_EvapCoolerFouling;
-                    }
+                // Check whether the evaporative cooler  name and type match each other;
+                EvapCoolerNum = UtilityRoutines::FindItemInList(state.dataFaultsMgr->FaultsEvapCoolerFouling(jFault_EvapCoolerFouling).EvapCoolerName,
```

Comment:
There is a lot of redundancy in this code because neither `Chillers` not `Faults` are leveraging inheritance properly. This is probably not the right place to do this for Chillers, but the way to start doing this for `Faults` would be to replace the 12 `Array1D<FaultsManager::FaultPropertiesX>` with `EPVector<FaultsManager::FaultProperties*>`. This will allow you to do all the lookups in a single array rather than in 12 separate arrays.